### PR TITLE
Updated the changelog for 2.5

### PR DESCRIPTION
**Pull request by Foaly**
I've went through the commit history and updated the changelog up to the commit 8b7a50a914d4986477081e6f94b676e6089180dd.

@eXpl0it3r I don't know what your criteria was in the past for including or skipping a commit. I descided to try to keep the changelog complete and include everything. I grouped smaller commits into one entry. But I suppose if the changelog is to long there is room to shorten it further.

The general block in the beginning turned mostly into a `Build` block, so maybe we should rename it as such or make a new section. What do you think?

Reviews or comments welcome 🙂 

PS: I would keep this open until we do a commit freeze, then update this to the latest commit and then merge.

### CODE REVIEW

**Comment by LaurentGomila** [comment]
> I descided to try to keep the changelog complete and include everything

Not sure that "removed dead code", "updated documentation" or "removed unused variable" is relevant in the changelog. People who want to see the full commit log can see it on with git (online on github if they haven't cloned SFML), so I would rather make this changelog focus on relevant stuff, ie. changes that matter to the end users.

I also see that you've added links to issues and PRs in the whole changelog, maybe this should be in a separate PR as it as nothing to do with the 2.5 changelog.

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,111 @@
...
+
+**Features**
+
+  * Implemented Cursor API ([#827](https://github.com/SFML/SFML/pull/827))
+  * Implemented Clipboard API ([#1221](https://github.com/SFML/SFML/pull/1221))
+  * [Windows] Added support for interfacing with joysticks via DirectInput when it is available ([#1326](https://github.com/SFML/SFML/pull/1326))
+
+**Bugfixes**
+
+  * Updated documentation ([#1255](https://github.com/SFML/SFML/pull/1255), [#1310](https://github.com/SFML/SFML/pull/1310))
```
Comment (eXpl0it3r) [inline changelog.md]: Don't need this in the changelogs

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,111 @@
...
+  * [macOS] Fixed window menu not working ([#1193](https://github.com/SFML/SFML/pull/1193))
+  * [macOS] Removed unused variable ([#1276](https://github.com/SFML/SFML/pull/1276))
```
Comment (eXpl0it3r) [inline changelog.md]: Refactors don't need to be in the changelog unless they have an impact on the API

File: changelog.md [hunk 3]
```diff
@@ -1,5 +1,111 @@
...
+
+**Bugfixes**
+
+  * Fixed typo in assert in `sf::Texture::update()` ([#1210](https://github.com/SFML/SFML/pull/1210))
+  * Properly free memory in `sf::Font::cleanup()` ([#1119](https://github.com/SFML/SFML/pull/1119))
+  * Fixed memory leak in `sf::Font` ([#1216](https://github.com/SFML/SFML/pull/1216))
+  * Fix OpenGL texture coordinate pointer not being updated correctly under certain conditions ([#1297](https://github.com/SFML/SFML/pull/1297))
+  * Fix for broken text when the font is reloaded ([#1345](https://github.com/SFML/SFML/pull/1345))
+  * Fix memory leak in `sf::Text` ([#1360](https://github.com/SFML/SFML/pull/1360))
+  * Refactoring of `sf::Text` ([#928](https://github.com/SFML/SFML/pull/928))
```
Comment (eXpl0it3r) [inline changelog.md]: This description and PR don't match. Also already mentioned above. Picked the wrong number?

File: changelog.md [hunk 4]
```diff
@@ -1,5 +1,111 @@
...
+  * Updated documentation ([#1197](https://github.com/SFML/SFML/pull/1197))
```
Comment (eXpl0it3r) [inline changelog.md]: Documentation changes shouldn't be in the changelog

File: changelog.md [hunk 5]
```diff
@@ -1,5 +1,111 @@
...
+  * [Windows] Replaced time-based joystick poll with a hardware event handler ([#1195](https://github.com/SFML/SFML/pull/1195), [#1179](https://github.com/SFML/SFML/pull/1179))
+  * [macOS] Worked around render target bug ([#1342](https://github.com/SFML/SFML/pull/1342))
+
+### Audio
+
+**Features**
+
+  * Added loop point support to `sf::Music` ([#629](https://github.com/SFML/SFML/pull/629))
+  * Added support for extensible format PCM wave files ([#1296](https://github.com/SFML/SFML/pull/1296))
```
Comment (eXpl0it3r) [inline changelog.md]: Maybe "Added support for the extensible PCM wave file format" instead

File: changelog.md [hunk 6]
```diff
@@ -1,5 +1,111 @@
...
+  * [iOS] Enable audio module ([#1338](https://github.com/SFML/SFML/pull/1338))
+
+**Bugfixes**
+
+  * Allow polymorphism with `sf::SoundSource` ([#1185](https://github.com/SFML/SFML/pull/1185))
+  * Fixed WAV file writer writing wrong header values ([#1281](https://github.com/SFML/SFML/pull/1281))
+  * Small bugfix to argument of `alcCaptureOpenDevice()` ([#1305](https://github.com/SFML/SFML/pull/1305))
+  * Removed dead code ([#1261](https://github.com/SFML/SFML/pull/1261))
+  * Updated documentation ([#1321](https://github.com/SFML/SFML/pull/1321), [#1353](https://github.com/SFML/SFML/pull/1353))
```
Comment (eXpl0it3r) [inline changelog.md]: See above (docs)

File: changelog.md [hunk 7]
```diff
@@ -1,5 +1,111 @@
...
+  * Added loop point support to `sf::Music` ([#629](https://github.com/SFML/SFML/pull/629))
+  * Added support for extensible format PCM wave files ([#1296](https://github.com/SFML/SFML/pull/1296))
+  * [iOS] Enable audio module ([#1338](https://github.com/SFML/SFML/pull/1338))
+
+**Bugfixes**
+
+  * Allow polymorphism with `sf::SoundSource` ([#1185](https://github.com/SFML/SFML/pull/1185))
+  * Fixed WAV file writer writing wrong header values ([#1281](https://github.com/SFML/SFML/pull/1281))
+  * Small bugfix to argument of `alcCaptureOpenDevice()` ([#1305](https://github.com/SFML/SFML/pull/1305))
+  * Removed dead code ([#1261](https://github.com/SFML/SFML/pull/1261))
```
Comment (eXpl0it3r) [inline changelog.md]: Not needed here

File: changelog.md [hunk 8]
```diff
@@ -1,5 +1,111 @@
...
+  * [macOS] Added support for (some) Hat/POV axis ([#1248](https://github.com/SFML/SFML/pull/1248))
+  * [Linux] Fixed `sf::Window::getPosition()` returning incorrect position because of differences in window managers ([#1228](https://github.com/SFML/SFML/pull/1228), [#1266](https://github.com/SFML/SFML/pull/1266))
+  * [Linux] Fix X11 key repeat handling not filtering out events from other windows ([#1291](https://github.com/SFML/SFML/pull/1291), [#1223](https://github.com/SFML/SFML/pull/1223), [#1230](https://github.com/SFML/SFML/pull/1230))
+  * [Linux] Restore fullscreen of a non-visible window ([#1339](https://github.com/SFML/SFML/pull/1339))
+  * [iOS] Fix build error when using @import ([#1272](https://github.com/SFML/SFML/pull/1272))
+  * [iOS] Don't need to find vorbisfile or vorbisenc ([#1347](https://github.com/SFML/SFML/pull/1347))
+  * [Windows] Prevent uninitialized read by zeroing memory ([#1264](https://github.com/SFML/SFML/pull/1264))
+  * [Windows] Fixed modifier keys handling ([#1357](https://github.com/SFML/SFML/pull/1357))
+  * [Android] Return correct key code for delete/backspace ([#1362](https://github.com/SFML/SFML/pull/1362))
```
Comment (eXpl0it3r) [inline changelog.md]: OS labels used to be ordered alphabetically (Android, iOS, Linux, macOS, Windows).

**Comment by eXpl0it3r** [comment]
Superseded by #1427

**Pull request closed without merging**